include/os/freebsd/spl/sys/time.h

```diff
@@ -62,6 +62,23 @@ typedef longlong_t	hrtime_t;
 #define	SEC_TO_TICK(sec)	((sec) * hz)
 #define	NSEC_TO_TICK(nsec)	((nsec) / (NANOSEC / hz))
 
+static __inline hrtime_t
+getlrtime(void)
+{
+	struct timespec ts;
+	hrtime_t nsec;
+
+	getnanotime(&ts);
```

Comment:
You've missed "up" here.  Without "up" that means time from epoch, not from boot, as returned by `nanouptime()` in `gethrtime()`.   Should be `getnanouptime(&ts)` here.